src/OMSimulatorLua/OMSimulatorLua.c

```diff
@@ -892,6 +892,11 @@ static int OMSimulatorLua_oms_addSubModel(lua_State *L)
 
   lua_pushinteger(L, status);
 
+  if (status!=oms_status_ok)
+  {
+    return luaL_error(L, "oms_addSubModel(%s,%s) failed", cref, fmuPath);
+  }
```

Comment:
I guess this should come before pushing the return value `status`.

Reply:
I don't understand what `lua_pushinteger` is supposed to do.  It pushes the error status to the first position in some Lua stack? So it will save the error and the `luaL_error` displays the error massage and aborts?

Reply:
The stack contains all return values, and the returned integer specifies the number of returned values, i.e. size of the stack.

Reply:
I moved it up